src/AuthenticatedAPIClient/axiosConfig.js

```diff
@@ -78,10 +78,11 @@ function applyAxiosInterceptors(authenticatedAPIClient) {
           queueRequests = false;
           PubSub.publishSync(ACCESS_TOKEN_REFRESH);
         })
-        .catch(() => {
-          // TODO: We should give the client app an opportunity to
-          // take control here before logout/redirect to sign in.
-          authenticatedAPIClient.logout();
+        .catch((error) => {
+          if (!authenticatedAPIClient.handleRefreshAccessTokenFailure
+            || authenticatedAPIClient.handleRefreshAccessTokenFailure(error)) {
```

Comment:
If the handler returns true we will logout?  This seems confusing and unnecessary.  Could this just be:
```
if (authenticatedAPIClient.handleRefreshAccessTokenFailure) {
  authenticatedAPIClient.handleRefreshAccessTokenFailure(error);
} else {
  authenticatedAPIClient.logout();
}
```
Maybe this is a common pattern I am not aware of, but I don't love it. :)

Reply:
fair enough :P I've seen it, but I suppose it's not exactly clear.